packages/@aws-cdk/aws-efs/lib/access-point.ts

```diff
@@ -109,23 +115,104 @@ export interface AccessPointProps extends AccessPointOptions {
   readonly fileSystem: IFileSystem;
 }
 
+/**
+ * Attributes that can be specified when importing an AccessPoint
+ */
+export interface AccessPointAttributes {
+  /**
+   * The ID of the AccessPoint
+   * One of this, of {@link accessPointArn} is required
+   *
+   * @default - no access point id
+   */
+  readonly accessPointId?: string;
+
+  /**
+   * The ARN of the AccessPoint
+   * One of this, of {@link accessPointId} is required
+   *
+   * @default - no access point arn
+   */
+  readonly accessPointArn?: string;
+
+  /**
+   * The efs filesystem
+   *
+   * @default - no efs filesystem
+   */
+  readonly fileSystem: IFileSystem;
+}
+
+abstract class AccessPointBase extends Resource implements IAccessPoint {
+  /**
+   * The ARN of the Access Point
+   * @attribute
+   */
+  public abstract readonly accessPointArn: string;
+
+  /**
+   * The ID of the Access Point
+   * @attribute
+   */
+  public abstract readonly accessPointId: string;
+
+  /**
+   * The filesystem of the access point
+   */
+  public abstract readonly fileSystem: IFileSystem;
+}
+
 /**
  * Represents the AccessPoint
  */
-export class AccessPoint extends Resource implements IAccessPoint {
+export class AccessPoint extends AccessPointBase {
   /**
    * Import an existing Access Point
    */
-  public static fromAccessPointId(scope: Construct, id: string, accessPointId: string): IAccessPoint {
+  public static fromAccessPointAttributes(scope: Construct, id: string, attrs: AccessPointAttributes): IAccessPoint {
     class Import extends Resource implements IAccessPoint {
```

Comment:
we should extract this to a class and return that type - here's an example in rds [`fromDatabaseClusterAttributes`](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-rds/lib/cluster.ts#L463-L465) for [`ImportedDatabaseCluster`](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-rds/lib/cluster.ts#L368) for inspiration